### DISCUSSION TITLE
Configuring Tailnet routing on your local network

### DISCUSSION THREAD

**Discussion by tomvoss** [SierraSoftworks/tailscale-udm#51] [Ideas]
There a few small tweaks that can be made to allow routing to/from other machines on the local network and account for multiple WAN links failing over.

1. Modify /etc/default/tailscaled by adding a socket and removing the userspace-networking.

```
FLAGS="--port 41641 --socket /var/run/tailscale/tailscaled.sock --state /data/tailscale/tailscaled.state"
```

2. Add local networks to tailscale's routing table (table 52) to avoid traffic being wrongly sent over the tailscale0 interface.

```
ROUTES_TO_ADD=$( ip route | grep "dev br" )
echo "${ROUTES_TO_ADD}" | while read -r route; do /sbin/ip route add ${route} table 52; done
```

3. Use a script similar to this one that ensures tailscale packets go out of the right WAN interface in the case that multiple WAN links are present: https://github.com/tailscale/tailscale/issues/4038#issuecomment-1091851637

```
#!/bin/sh

RULE_PRIORITY="5225"
SLEEP_INTERVAL="0.25"
TABLE=-1

getDefaultRouteTable() {
    /sbin/ip rule list priority 32766 | cut -d " " -f 4
}

updateTailscaleMarkingRule() {
    #if default route table changed then update ip rules accordingly
    if [ ! $TABLE -eq $1 ] && [ $1 -gt 0 ]
    then
            /sbin/ip rule del priority $RULE_PRIORITY
            /sbin/ip rule add priority $RULE_PRIORITY from all fwmark 0x80000 lookup $1

            TABLE=$1
    fi
}

until false; do
    updateTailscaleMarkingRule $(getDefaultRouteTable)
    
    sleep $SLEEP_INTERVAL
done
```

4. Add routes and exit nodes as necessary.

```
tailscale up --advertise-exit-node --advertise-routes="192.168.0.0/24,192.168.1.0/24" --snat-subnet-routes=false --accept-routes --reset
```

**Comment by notheotherben** [comment]
Hi @tomvoss, thanks for bringing this information to my attention. I assume that you've tested and are running this on UniFi OS 1.x (i.e. a 1.12.x firmware version), but looking at the backing thread in tailscale/tailscale#4038 it sounds like this is something that Tailscale should be configuring internally.

Given the need to setup and run a separate process (in this case a script) to constantly ensure that the correct routing table is configured and the requirement to manually tweak route rules, I'm going to opt to not include this change in the core tailscale-udm scripts (as we've worked around this problem by using userspace networking).

What I will do is convert this issue into a discussion and link it in the README under the section relating to exposing your Tailnet via your UDM (i.e. not requiring Tailscale to be installed on devices within your local network). I'll also be following that Tailscale thread to see what (if any) updates come up so that I can make adjustments here as needed.

**Comment by jonp92** [comment]
Just FYI I tried this on a UDM pro with 1.x firmware and it doesn't work. In fact Tailscale won't even connect at all with those settings.

**Comment by austin-millan** [comment]
Disclaimer: I'm new to Tailscale having only set it up yesterday, and only somewhat familiar with Wireguard.

I wasn't able to get it working either with the above configurations. However, I did get it working by following the instructions here: https://tailscale.com/kb/1019/subnets/#enable-ip-forwarding

And then going to the Tailscale UI and enabling the following flags under "Edit Route Settings" for the router/node.

- Connect machines you can’t install Tailscale on.
- Allow your network to route internet traffic through this machine.

**Reply by kevinsegal** [reply]
> https://tailscale.com/kb/1019/subnets/#enable-ip-forwarding

This config worked like a charm. Should add this to the main documentation, this is something a lot of people (including myself) have been trying to do for a long time, and tail scale just managed to do what zerotier couldn't do for me.

**Reply by roydufek** [reply]
May I ask what exact steps you took to make this work? I can route to the networks inside my UDM from my android phone but I can't then route to other advertised routes outside my UDM. Not sure what step I missing... I did add the ip forwarding via the 99-tailscale.conf file but not sure how to finish it off with the firewall step as the UDM doesn't have the "firewalld". I'm pretty newbie when it comes to all this but just trying to get the UDM to act like any other linux router type installation would but don't understand userspace networking all that much. Were these instructions adding to the changes suggested above or did you install and only follow enable IP forwarding? Thank you for any help you can provide! I've spent hours banging my head on wall trying to make this work... I think I am missing the masquerade step but not sure how to add that for the UDM and tailscale setup. Thanks again- :)

**Comment by FearNaBoinne** [comment]
I am on UnifyOS v3.0.20.

I got a message that the route already existed (`RTNETLINK answers: File exists`) and had to first delete it before I could add it to table 52.
When I did that, I could add the rule and have access over both the TailScale SSH console and the local SSH.

So my script (quick and dirty) looks like:
```
ROUTES_TO_ADD=$( ip route | grep "dev br" )
ROUTES_TO_DELETE=$( ip route | grep "dev br" |  cut -d " " -f 1)
echo "${ROUTES_TO_DELETE}" | while read -r route; do /sbin/ip route del ${route} table 52; done
echo "${ROUTES_TO_ADD}" | while read -r route; do /sbin/ip route add ${route} table 52; done
```

That got the connection up (together with the priority rule to allow tailscale to route out of the correct WAN port), however it doesn't resolve *all* my problems.

I can ping remote tailscale IPs _and the networks behind them_ from the UDM Pro shell, I can ping the local tailscale IP from my hosts, but I can**not** ping the remote tailscale IPs _from my hosts_, let alone the networks behind them.

_PS: I am not certain the ROUTES_TO_DELETE part works correctly if there are multiple IP subnets to route locally, because I am pretty certain this will only give me the first one, but I haven't been able to test it, because I only have the one subnet right now!_

**Comment by jhavens12** [comment]
This doesn't work for me - nothing happens when I follow the instructions. I can't even ping any of the tailscale machine IPs from UDM. What am I doing wrong?

**Reply by ksc98** [reply]
Same.  At this point whoever wants a fancy cup of coffee, hit me up! :)

**Comment by jasonwbarnett** [comment]
@tomvoss this worked wonderfully. Thank you

I made one modification to the script though.

```sh
getDefaultRouteTable() {
    /sbin/ip rule list priority 32766 | cut -d " " -f 4 | cut -d "." -f 1
}
```

The output of `/sbin/ip rule list priority 32766 | cut -d " " -f 4` on my UDM-Pro was:

```
$ /sbin/ip rule list priority 32766 | cut -d " " -f 4
202.eth9
```

**Reply by jasonwbarnett** [reply]
I also forked tailscale to eliminate the script completely. Though, my fork doesn't automate failover. This is my home network and I only have a single WAN connection.

https://github.com/jasonwbarnett/tailscale/tree/jwb/add-support-for-udm-pro

**Reply by jasonwbarnett** [reply]
I also overrode the systemd `ExecStart` so that arguments weren't being double up. I'm not sure if it has any actual impact, but it seemed cleaner to me.

```sh
$ EDITOR=vim systemctl edit tailscaled.service
```

```text
### Editing /etc/systemd/system/tailscaled.service.d/override.conf
### Anything between here and the comment below will become the new contents of the file

[Service]
ExecStart=
ExecStart=/usr/sbin/tailscaled --port=${PORT} $FLAGS

### Lines below this comment will be discarded

### /lib/systemd/system/tailscaled.service
# [Unit]
# Description=Tailscale node agent
# Documentation=https://tailscale.com/kb/
# Wants=network-pre.target
# After=network-pre.target NetworkManager.service systemd-resolved.service
#
# [Service]
# EnvironmentFile=/etc/default/tailscaled
# ExecStartPre=/usr/sbin/tailscaled --cleanup
# ExecStart=/usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=${PORT} $FLAGS
# ExecStopPost=/usr/sbin/tailscaled --cleanup
#
# Restart=on-failure
#
# RuntimeDirectory=tailscale
# RuntimeDirectoryMode=0755
# StateDirectory=tailscale
# StateDirectoryMode=0700
# CacheDirectory=tailscale
# CacheDirectoryMode=0750
# Type=notify
#
# [Install]
# WantedBy=multi-user.target
```

```text
$ cat /etc/default/tailscaled
# Set the port to listen on for incoming VPN packets.
# Remote nodes will automatically be informed about the new port number,
# but you might want to configure this in order to set external firewall
# settings.
PORT="41641"

# Extra flags you might want to pass to tailscaled.
FLAGS="--state=/data/tailscale/tailscaled.state --socket=/var/run/tailscale/tailscaled.sock"
```

```
$ ps -ef | grep tailscale[d]
root     2496726       1  0 Dec05 ?        02:15:39 /usr/sbin/tailscaled --port=41641 --state=/data/tailscale/tailscaled.state --socket=/var/run/tailscale/tailscaled.sock
```

**Comment by LP0101** [comment]
I'm currently thinking about buying a UDM-pro to replace my firewall. What's the current status on this on the latest UnifyOS with only one WAN connection? Does following the instructions [here](https://tailscale.com/kb/1019/subnets?tab=linux) work? Is there more that needs to be done?

**Reply by jasonwbarnett** [reply]
@LP0101 You still need to use this project to install tailscale on a UDM-Pro and if you don't want to install tailscale on each device then you would need to follow the directions in this discussion outlined by @tomvoss

**Reply by CaptInsano** [reply]
Sorry to ask this again but there has been discussion in this thread about improvements to the excellent work by @tomvoss at the start of this thread by other users so I am confused:

After installing tailscale as per this repo, is the first post of this thread still the best option to follow in order to allow devices on my LAN **without** tailscale to access other devices/subnets on my tailnet? Or do I just install tailscale and then follow the instructions of https://tailscale.com/kb/1019/subnets?tab=linux ?

@jasonwbarnett You had mentioned improvements to script & even removal of the need for the script in your own fork but I cannot see how to install same on my UDM.

Thanks to everyone of their work on this!

**Reply by jasonwbarnett** [reply]
@CaptInsano I don't hold myself as the authority here, so take what I say with a grain of salt.

> After installing tailscale as per this repo, is the first post of this thread still the best option to follow in order to allow devices on my LAN without tailscale to access other devices/subnets on my tailnet? Or do I just install tailscale and then follow the instructions of https://tailscale.com/kb/1019/subnets?tab=linux ?

Yes, this discussion is still the most up-to-date information for configuring a UDM-Pro as a subnet router.

> @jasonwbarnett You had mentioned improvements to script & even removal of the need for the script in your own fork but I cannot see how to install same on my UDM.

1. I never tried following @austin-millan comment.
2. The code I wrote in my fork is really hacky and probably won't ever be merged until I work with someone over at tailscale to figure out a good way to do what I did that doesn't interfere with other tailscale use cases. I need to pick this up, so thanks for the nudge.

I don't know how technical you are, but if you're familiar with go it's pretty straight forward to compile tailscale off of my branch and upload it, replacing the binaries with the ones compiled.

A high level looks like this (this should be done *after* following the standard `SierraSoftworks/tailscale-udm` installation):

```sh
# Clone Repo
git clone -b jwb/add-support-for-udm-pro https://github.com/jasonwbarnett/tailscale.git

# Compile binaries
cd tailscale
./build_udm_pro.sh

# Upload binaries to UDM-Pro
scp ./tailscale{,d} root@<ip-of-udm-pro>:~/
```

```sh
# Login via SSH to the UDM-Pro
ssh root@<ip-of-udm-pro>

# Overwrite original binaries with new ones
mv ./tailscaled /usr/sbin/tailscaled
mv ./tailscale /usr/bin/tailscale

# Restart tailscaled
systemctl restart tailscaled
```

You can confirm everything worked by visiting https://login.tailscale.com/admin/machines and the version reported for the UDM-Pro should be: `1.56.1-udm-pro`

![Screenshot 2024-01-11 at 5 14 11 PM](https://github.com/SierraSoftworks/tailscale-udm/assets/862300/b8fb9059-5841-4b11-bb10-42f036bc263f)

**Reply by jasonwbarnett** [reply]
tailscale PR opened: https://github.com/tailscale/tailscale/pull/10828

**Reply by mickg10** [reply]
@jasonwbarnett Just to check - as that PR is still pending - does this still require user-space networking? Somehow, after upgrading to 1.64, not having use-space networking broke with `tailscaled` outputting: 
`Received error: fetch control key: Get "https://controlplane.tailscale.com/key?v=90": dial tcp [2a05:d014:386:202:4d:f871:20f3:6602]:443: connect: network is unreachable`

I do wonder if they broke something...

**Comment by jasonhollis** [comment]
This all worked perfectly for me but I'm still getting:

tailscale status
100.122.23.76   ktp-newport          jason@       linux   -
100.84.226.19   brainassistant       jason@       linux   idle; offers exit node; offline
100.119.51.50   hanewport            jason@       linux   idle; offers exit node; offline
100.76.248.120  haosnewport          jason@       linux   idle; offers exit node; offline
100.127.227.41  hapi5-1              jason@       linux   idle; offers exit node; offline
100.120.49.71   homeassistant        jason@       linux   idle; offers exit node
100.104.22.108  jfh14m2max-2         jason@       macOS   offline
100.119.177.51  jfh16m1max-2         jason@       macOS   offline
100.87.136.8    jfhipadpro           jason@       iOS     offline
100.95.146.87   jfhiphone14promax    jason@       iOS     offline
100.79.125.130  mabels-macbook-pro   mabel@       macOS   offline

# Health check:
#     - Update available: 1.56.1 -> 1.62.1, run `tailscale update` or `tailscale set --auto-update` to update
#     - dns-os: getting OS base config is not supported
#     - dns: getting OS base config is not supported
root@KTP-Newport:~# ping 100.120.49.71
PING 100.120.49.71 (100.120.49.71) 56(84) bytes of data.

--- 100.120.49.71 ping statistics ---
10 packets transmitted, 0 received, 100% packet loss, time 9332ms

root@KTP-Newport:~# tailscale ping 100.120.49.71
pong from homeassistant (100.120.49.71) via 103.214.220.33:45494 in 11ms

It's not working!  It seems totally crazy to me to have to put Tailscale in other places when I have a perfect UDMP SE sitting there were it should be running!

**Comment by jasonwbarnett** [comment]
Great news! The PR I opened ([tailscale/tailscale#10828](https://github.com/tailscale/tailscale/pull/10828)) has been merged. Once this makes it into a stable Tailscale release, we’ll no longer need to use the script. Instead, we can simply modify `/etc/default/tailscaled` to achieve the same result. 🎉

I opened a separate issue to discuss the possibility of modifying the installer so we wouldn't need to modify the `/etc/default/tailscaled` file post install. https://github.com/SierraSoftworks/tailscale-udm/issues/98

Based on the recent stable release history:
* 1.72: Aug 19
* 1.74: Sep 12
* 1.76: Oct 10
* 1.78: Dec 5

It seems likely the next stable release (1.80) will be out shortly!

**Reply by notheotherben** [reply]
That's awesome news, I did unfortunately see that the implementation of tailscale/tailscale#10828 appears to only include support for the UDM Pro (i.e. other UDM variants are not supported), but the fact that this is present in the first place is a great step in the right direction - nice work getting it included in Tailscale proper.

**Reply by jasonwbarnett** [reply]
@notheotherben If you and I could spend time together, I bet I could get another PR merged upstream in no time. I'm lacking the experience on the UDM side. Do you hang out in any discord servers or slack communities?

**Reply by notheotherben** [reply]
Happy to do so - afraid the only Discords I hang out in are related to 3D Printing (VORONDesign, prusa3d), but I am happy to hop into another if that works for you - username there is the same as here.

**Reply by Menschomat** [reply]
Cann you give us a short step by step guide on what do do now to get things running on UDM with your ne PR?
Only the changes to /etc/default/tailscaled mentioned in the first Post?

**Reply by jasonwbarnett** [reply]
@Menschomat I'll post something by the time this is actually released into a stable tailscale release which I anticipate will happen in the next 60 days.

**Reply by jasonwbarnett** [reply]
If you'd like to test this out, it's officially been released in tailscale's [*unstable track*][1]. These instructions assume you've already installed tailscale using `github.com/SierraSoftworks/tailscale-udm` installer.

1. Add Tailscale's GPG key

   ```sh
   sudo mkdir -p --mode=0755 /usr/share/keyrings
   curl -fsSL https://pkgs.tailscale.com/unstable/debian/bullseye.noarmor.gpg | sudo tee /usr/share/keyrings/tailscale-archive-keyring.gpg >/dev/null
   ```

2. Add the tailscale repository

   ```sh
   curl -fsSL https://pkgs.tailscale.com/unstable/debian/bullseye.tailscale-keyring.list | sudo tee /etc/apt/sources.list.d/tailscale.list
   ```

3. Install/Update Tailscale

   ```sh
   sudo apt-get update && sudo apt-get install tailscale=1.79.101
   ```

4. Edit `/etc/default/tailscaled`

   ```sh
   cat <<'EOF' | sudo tee /etc/default/tailscaled > /dev/null
   # Set the port to listen on for incoming VPN packets.
   # Remote nodes will automatically be informed about the new port number,
   # but you might want to configure this in order to set external firewall
   # settings.
   PORT="41641"
   
   # Extra flags you might want to pass to tailscaled.
   FLAGS="--state=/data/tailscale/tailscaled.state --socket=/var/run/tailscale/tailscaled.sock"
   
   EOF
   ```

5. Start Tailscale!

   ```sh
   # Start tailscaled
   sudo tailscale up --advertise-routes="<your-local-subnet-cidr>" --snat-subnet-routes=false --accept-routes --reset
   
   # i.e.
   # sudo tailscale up --advertise-routes="10.0.0.0/24" --snat-subnet-routes=false --accept-routes --reset
   ```

[1]: https://pkgs.tailscale.com/unstable/?v=1.79.101#debian-bullseye

**Reply by Firaenix** [reply]
This works great for me.

I run nextdns on my UDM SE alongside Tailscale.

I have set up nextdns to forward queries about ts.net domains to 100.100.100.100 and have this systemd unit set up to run at startup:

```
cat << 'EOF' > /etc/systemd/system/tailscale-network.service
[Unit]
Description=Tailscale Network Configuration
After=network.target tailscaled.service

[Service]
Type=oneshot
ExecStart=/bin/sh -c '\
    ip route del 100.64.0.0/10 dev tailscale0 2>/dev/null || true && \
    ip route add 100.64.0.0/10 dev tailscale0 && \
    iptables -A FORWARD -i br0 -o tailscale0 -j ACCEPT && \
    iptables -A FORWARD -i tailscale0 -o br0 -j ACCEPT && \
    iptables -t nat -A POSTROUTING -o tailscale0 -j MASQUERADE'
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
EOF

systemctl daemon-reload
systemctl enable tailscale-network
systemctl start tailscale-network
```

Now I can access all my Tailscale IPs on my local network without having Tailscale installed as well as all MagicDNS addresses!

Only issue I do seem to have is that when my UDM SE reboots and upgrades its firmware, I have to reinstall tailscale (tailscale=1.79.130) along with the unstable repositories & config for /etc/default/tailscaled as specified above, restart the tailscaled systemd unit and then re run my tailscale-network systemd unit.

Other than that, its fantastic

**Reply by jasonwbarnett** [reply]
I have the same issue. I actually disabled automatic updates so I don't lose it randomly since my whole house DNS is served on my tailnet.

**Reply by FidikVien** [reply]
Just wanted to check in on this, as it appears 1.80.2 is in the stable tailscale channel. However following your guide a from Jan 15, I still am only able to ping accepted routes from the UDM, not from any downstream client. I did follow the guide at the top from 2/2023, including the routes to add, could these be conflicting now? Thanks for your efforts on this.

**Reply by Menschomat** [reply]
Yeah an "official" "final" guide would be nice, since 1.80.2 is now live 👍🏻

**Reply by jasonwbarnett** [reply]
I'm traveling for work this week. When I get back I'd like to write up a final guide and get it merged into main.

**Reply by jasonwbarnett** [reply]
Official docs authored: https://github.com/SierraSoftworks/tailscale-udm/pull/105

**Reply by FidikVien** [reply]
Hey Jason,

Thanks for updating the thread about this. Two things I'd noticed implementing what you showed here. I can confirm that I am no longer in userspace networking. However when I run a cat on /var/run/tailscale/tailscaled.sock I see this file as blank?  I am able to ping tailscale subnet IPs from the UDM but not actual clients. Is there a firewall or NAT entry that is required? I am running the new Zone-based firewall.

I also noticed when I did the tailscale up after going through this process everything on the LAN side was unreachable, the UDM showed in unifi.ui.com over an LTE connection, but then the LAN just started working again... somewhat disturbing.

**Reply by notheotherben** [reply]
@FidikVien `cat /.../tailscaled.sock` won't give anything useful as its a unix socket rather than a plain file - so that's entirely normal.

As for traffic routing, after following the instructions (and also running the new zone-based firewall) I'm seeing the same symptoms. I played around a bit in an attempt to get stuff working, but to no avail - I'll try again tomorrow and report back if I'm able to find a solution.

(No issues with network disconnects when applying the changes)

**Reply by FidikVien** [reply]
I did think that was a possibility, and with all the attempts at getting my install working correctly before everyone's recent work, I removed tailscale and just performed a reinstall. I can confirm I'm on 1.80.2 now, however we're back where we were before, pinging remotes from the UCG, but not from clients on the LAN. Is there something firewall related going on here? NAT?

Thanks for your response!

**Reply by jasonwbarnett** [reply]
## Debugging Tailscale on UDM-Pro

Today, I had a bit of a head-scratcher trying to figure out why Tailscale wasn’t working on my UDM-Pro after an upgrade. After reinstalling Tailscale, I found that communication in both directions was completely broken.

Turns out, I had forgotten to approve both the published routes and the exit node setting in the Tailscale admin portal. 🤦‍♂️ Once I accepted these, everything worked perfectly.

I diagnosed this issue using:

```sh
tcpdump -i tailscale0 icmp
```

By running this on my UDM-Pro and pinging a Tailscale client from a non-tailscale client device on my home network, I could see packets leaving the `tailscale0` interface on my UDM-Pro, but no response. This led me to investigate why packets weren’t flowing back, which is when I realized that after wiping and reinstalling the Tailscale client, I needed to re-approve the routes in the Tailscale admin portal.

## Important Note for Linux Users

Although this wasn’t an issue for me (since it was already set up), Linux clients must explicitly accept subnet routes. If not, they won’t be added. To do this, run:

```sh
sudo tailscale up --accept-routes
```

More details here: [Tailscale Subnet Routing](https://tailscale.com/kb/1019/subnets#use-your-subnet-routes-from-other-devices)

Hope this helps someone avoid a similar headache!

**Reply by ZeidAqemia** [reply]
> This works great for me.
> 
> I run nextdns on my UDM SE alongside Tailscale.
> 
> I have set up nextdns to forward queries about ts.net domains to 100.100.100.100 and have this systemd unit set up to run at startup

@Firaenix Could you please share the configuration for your nextdns + Tailscale setup or explain what you had to do?
I did it too and I can't ping ts.net subdomains from local clients and I don't know what I'm missing.

**Reply by cristiangauma** [reply]
Thank you very much!. This worked perfectly in order to have a site-to-site with --snat-subnet-routes=false and multiple VLANs.

**Reply by GreenEnvy22** [reply]
I'm supporting someone who has had a few sites all with older EdgeRouters which I had installed tailscale on, and was working fine as subnet routers for their 3 sites.
I recently replaced one Edgerouter with a UDM Pro. 
I installed Tailscale using this repo, and after reading this discussion, got it working as a subnet router too.
I am having one issue though, the modifications to `/etc/default/tailscaled` get wiped out frequently, seemingly daily.
My file looks slightly different, with there being a `PORT=41641` line, then another `FLAGS=` line. That FLAGS line I modify to remove the userspace, and add in the socket.
I then restart the tailscale service, and now subnet routing is working.
The next day however, it's down again, and the config file is back to having the userspace flag, and no socket.
Device is running UniFi OS 4.3.6, and it's not rebooting, so this isn't just a startup thing.

Any thoughts?

**Reply by notheotherben** [reply]
The contents of `/etc/default` are subject to replacement whenever your UDM is updated, so this script sets up a `systemd` job which regularly ensures that Tailscale is installed and configured correctly - including replacing the contents of that file.

If you'd like to persist your changes, make sure you add them to the `/data/tailscale/tailscale-env` file (as shown [here](https://github.com/SierraSoftworks/tailscale-udm/blob/dd0b40ce4473a7cbeb3b6fc7cb7f2623410020e4/package/tailscale-env#L2)). That should resolve the problem you're running into (and will ensure that your changes persist through OS upgrades etc).

**Comment by kwhelchel** [comment]
Total noob here and lost 
Ok i have tailscale running on a udm pro set as an exit node site A. This part i have running and able to use the tailscale windows app to connect to it with no issue. Using this project to accomplish this from sierrasoftworks tailscale-udm project on github.

Now what I need to have happen is on the other side site B is to be able to have it pass the tailscale site to site connection through the unifi device on site B. Site B is a UDR7 and I have it running the tailscale on it but can not make the connection from the device to site A.
